src/k8s-extension/azext_k8s_extension/partner_extensions/WorkloadIAM.py

```diff
@@ -44,6 +45,15 @@ def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
             # TODO - Set this to 'stable' when the extension is ready
             release_train = 'preview'
 
+        # The name is used as a base to generate Kubernetes labels for config maps, pods, etc, and
+        # their names are limited to 63 characters (RFC-1123). Instead of calculating the exact
+        # number of characters that we can allow users to specify, it's better to restrict that even
+        # more so that we have extra space in case the name of a resource changes and it pushes the
+        # total string length over the limit.
+        if len(name) > 20:
+            raise InvalidArgumentValueError(
+                f"Name '{name}' is too long, it must be 20 characters long max.")
```

Comment:
Is it worth declaring a constant and sharing this with the test code?

Reply:
If we do it here, we would have to do it for every string. Also, I'm not sure if f-strings would work in that case, which means that the change could be quite distracting. If we do that, we should do it as part of a different PR to simplify the review of this one.